cloud/http.go

```diff
@@ -0,0 +1,239 @@
+package cloud
```

Comment:
Not new code in this file, just move here from `client.go`.

Reply:
ty